boilerplate/app/app.tsx

```diff
@@ -26,10 +26,15 @@ import { ErrorBoundary } from "./screens/error/error-boundary"
 
 export const NAVIGATION_PERSISTENCE_KEY = "NAVIGATION_STATE"
 
+interface AppProps {
+  hideSplashScreen: () => Promise<void>
```

Comment:
Ideally, we import and use the splash-screen library API directly in this file. However, since we're using `expo-splash-screen` on Expo and `react-native-bootsplash` on vanilla, the lib is imported in the entry file and this method is exposed to be used there.